src/main/java/com/sublinks/sublinksapi/api/sublinks/v1/community/services/SublinksCommunityService.java

```diff
@@ -54,24 +60,21 @@ public CommunityResponse createCommunity(CreateCommunity createCommunity, Person
   public CommunityResponse updateCommunity(String key, UpdateCommunity updateCommunityForm,
       Person person) {
 
+    String domain = ActorIdUtils.getActorDomain(key);
+    if (domain != null && domain.equals(localInstanceContext.instance().getDomain())) {
+      key = ActorIdUtils.getActorId(key);
+    }
+    Optional<Community> foundCommunity = communityRepository.findCommunityByTitleSlug(key);
```

Comment:
Instead of loading the entire object you can simply check if it exists for better performance. Checkout https://www.baeldung.com/spring-data-exists-query

Reply:
It is required further down.